src/components/Card/MeetingRoom.tsx

```diff
@@ -0,0 +1,304 @@
+import React, { FC, useState, useEffect } from "react";
+import {
+    Box,
+    Typography,
+    IconButton,
+    Card,
+    CardContent,
+    Modal,
+    TextField,
+    Button,
+} from "@mui/material";
+import EditIcon from "@mui/icons-material/Edit";
+import DeleteForeverIcon from "@mui/icons-material/DeleteForever";
+import CloseIcon from "@mui/icons-material/Close";
+import axios from "axios";
+
+const styleEdit = {
+    position: "absolute" as "absolute",
+    top: "50%",
+    left: "50%",
+    transform: "translate(-50%, -50%)",
+    width: 600,
+    height: 500,
+    bgcolor: "background.paper",
+    borderRadius: 2,
+    boxShadow: 24,
+    p: 4,
+};
+
+const styleDelete = {
+    position: "absolute" as "absolute",
+    top: "50%",
+    left: "50%",
+    transform: "translate(-50%, -50%)",
+    width: 600,
+    height: 250,
+    bgcolor: "background.paper",
+    borderRadius: 2,
+    boxShadow: 24,
+    p: 4,
+};
+
+const MeetingRoom: FC<MeetingRoomProps> = ({
+    meetingRoomName,
+    meetingRoomID,
+    meetingRoomDescription,
+    latestBook,
+}) => {
+    const [editModalOpen, setEditModalOpen] = useState(false);
+    const [deleteModalOpen, setDeleteModalOpen] = useState(false);
+    const [editTitle, setEditTitle] = useState("");
+    const [editDescription, setEditDescription] = useState("");
+
+    useEffect(() => {
+        setEditTitle(meetingRoomName);
+        setEditDescription(meetingRoomDescription);
+    }, []);
+
+    const handleEditModalOpen = () => {
+        setEditModalOpen(true);
+    };
+
+    const handleEditModalClose = () => {
+        setEditModalOpen(false);
+    };
+
+    const handleDeleteModalOpen = () => {
+        setDeleteModalOpen(true);
+    };
+
+    const handleDeleteModalClose = () => {
+        setDeleteModalOpen(false);
+    };
+
+    const capacityFromDescription = () => {
+        const v = meetingRoomDescription.split(" ").reverse();
+        const ne = v.join().match(/[0-9]+/);
+        return ne;
+    };
+
+    const handleEdit = async () => {
+        console.log(meetingRoomID);
+        return await axios.patch(
+            "http://localhost:3001/rooms/" + meetingRoomID,
+            {
+                title: editTitle,
+                description: editDescription,
+            }
+        );
+    };
+
+    const handleDelete = async () => {
+        return await axios.delete(
+            "http://localhost:3001/rooms/" + meetingRoomID
+        );
+    };
+    return (
```

Comment:
This return is quite big, can you make it smaller by splitting it in functions or different components?